### PR TITLE
feat: [SRM-13376]: Adding API impl for improved verify step UX.

### PR DESCRIPTION
**Pull request by dhruv-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/42020)

### CODE REVIEW

File: srm-service/modules/cv-nextgen-service/contracts/api/src/main/java/io/harness/cvng/resources/VerifyStepResource.java [hunk 1]
```diff
@@ -69,7 +71,8 @@ public interface VerifyStepResource {
   PageResponse<MetricsAnalysis>
   getMetricsAnalysisForVerifyStepExecutionId(@BeanParam @Valid VerifyStepPathParams verifyStepPathParams,
       @QueryParam("anomalousMetricsOnly") @DefaultValue("false") boolean anomalousMetricsOnly,
-      @QueryParam("healthSource") List<String> healthSource, @QueryParam("node") List<String> node,
+      @QueryParam("healthSource") List<String> healthSource,
+      @QueryParam("transactionGroup") List<String> transactionGroup, @QueryParam("node") List<String> node,
```
Comment (n00bitax) [inline srm-service/modules/cv-nextgen-service/contracts/api/src/main/java/io/harness/cvng/resources/VerifyStepResource.java:75]: List<String> is called node is there any reason we didn't name it nodes or nodeList ?

File: srm-service/modules/cv-nextgen-service/contracts/commons/src/main/java/io/harness/cvng/cdng/beans/v2/ClusterAnalysisOverview.java [hunk 2]
```diff
@@ -12,7 +12,7 @@
 
 @Value
 @SuperBuilder
-public class ErrorsAnalysisOverview {
+public class ClusterAnalysisOverview {
   int knownClusters;
   int unknownClusters;
```
Comment (n00bitax) [inline srm-service/modules/cv-nextgen-service/contracts/commons/src/main/java/io/harness/cvng/cdng/beans/v2/ClusterAnalysisOverview.java:17]: my suggestion will be to suffix it with count.

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/analysis/services/impl/DeploymentTimeSeriesAnalysisServiceImpl.java [hunk 3]
```diff
@@ -146,6 +169,51 @@ public TimeSeriesAnalysisSummary getAnalysisSummary(List<String> verificationJob
...
+          healthy++;
+          break;
+        case OBSERVE:
+          warning++;
+          break;
+        case NEED_ATTENTION:
+          warning++;
+          break;
+        case UNHEALTHY:
+          unhealthy++;
```
Comment (n00bitax) [inline srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/analysis/services/impl/DeploymentTimeSeriesAnalysisServiceImpl.java:204]: we should throw an exception with default if it's guaranteed to match something.

File: srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/analysis/services/impl/DeploymentTimeSeriesAnalysisServiceImpl.java [hunk 4]
```diff
@@ -282,6 +350,197 @@ private List<TransactionMetricInfo> getMetrics(String accountId, String verifica
...
+        .analysisResult(AnalysisResult.fromRisk(hostData.getRisk()))
+        .analysisReason(getAnalysisReason(hostData))
+        .controlDataType(controlDataType)
+        .controlNodeIdentifier(hostData.getNearestControlHost())
+        .controlData(getMetricValuesFromRawValues(hostData.getControlData()))
+        .testData(getMetricValuesFromRawValues(hostData.getTestData()))
+        .build();
+  }
+
+  private static AnalysisReason getAnalysisReason(HostData hostData) {
```
Comment (n00bitax) [inline srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/analysis/services/impl/DeploymentTimeSeriesAnalysisServiceImpl.java:467]: too much code in this class, we should move all these static methods to a helper class.

File: srm-service/modules/cv-nextgen-service/contracts/api/src/main/java/io/harness/cvng/resources/VerifyStepResource.java (hunk 1 above)
Comment (n00bitax) [reply srm-service/modules/cv-nextgen-service/contracts/api/src/main/java/io/harness/cvng/resources/VerifyStepResource.java:75]: As per my understanding path param is mandatory by default, we need to put validations like not-null for query params if they are mandatory.

**Comment by n00bitax** [comment]
> **SonarQube Quality Gate**
> 
> [Quality Gate failed](https://sonar.harness.io/dashboard?id=harness-core-sonar-pr&pullRequest=42020)
> 
> [Failed condition](https://sonar.harness.io/dashboard?id=harness-core-sonar-pr&pullRequest=42020) 7 New Major Issues (is greater than 5)
> 
> [See analysis details on SonarQube](https://sonar.harness.io/dashboard?id=harness-core-sonar-pr&pullRequest=42020)
> 
> Fix issues before they fail your Quality Gate with [SonarLint](https://www.sonarqube.org/sonarlint/?referrer=pull-request) [SonarLint in your IDE](https://www.sonarqube.org/sonarlint/?referrer=pull-request).

That's very few sonar issues, well done.

**Review by dhruv-harness — COMMENTED** [review]
Thanks!

> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42020#-:-NKRgjNI1B3VB4bDUy1P:b-ucfknh)* status: 28 file(s) to review. 3. unresolved discussions. 0 of 1 approvals obtained. 

___
*[`srm-service/modules/cv-nextgen-service/contracts/api/src/main/java/io/harness/cvng/resources/VerifyStepResource.java` line 75 at r3](https://reviewable.io/reviews/harness/harness-core/42020#-NKKB-M3oHkYKQghJID--r3-75:-NKRgrc8AOGmBcEyUU7t:b-wdf6vt) ([raw file](https://github.com/harness/harness-core/blob/4b5095f490cdefcc1340afbacacec95a76a09a87/srm-service/modules/cv-nextgen-service/contracts/api/src/main/java/io/harness/cvng/resources/VerifyStepResource.java#L75)):*
<details><summary><i>Previously, n00bitax (Ansuman Satapathy) wrote…</i></summary><blockquote>

As per my understanding path param is mandatory by default, we need to put validations like not-null for query params if they are mandatory.
</blockquote></details>

Will refactor the parameters as suggested.
___
*[`srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/analysis/services/impl/DeploymentTimeSeriesAnalysisServiceImpl.java` line 204 at r3](https://reviewable.io/reviews/harness/harness-core/42020#-NKKB-M3oHkYKQghJIDC-r3-204:-NKRkSbc1EqEMymas5VN:b32w0x) ([raw file](https://github.com/harness/harness-core/blob/4b5095f490cdefcc1340afbacacec95a76a09a87/srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/analysis/services/impl/DeploymentTimeSeriesAnalysisServiceImpl.java#L204)):*
<details><summary><i>Previously, n00bitax (Ansuman Satapathy) wrote…</i></summary><blockquote>

we should throw an exception with default if it's guaranteed to match something.
</blockquote></details>

OK
___
*[`srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/analysis/services/impl/DeploymentTimeSeriesAnalysisServiceImpl.java` line 467 at r3](https://reviewable.io/reviews/harness/harness-core/42020#-NKKB-M3oHkYKQghJIDC-r3-467:-NKRkUG60FqbbuFhnPqF:b3vi6kr) ([raw file](https://github.com/harness/harness-core/blob/4b5095f490cdefcc1340afbacacec95a76a09a87/srm-service/modules/cv-nextgen-service/service/src/main/java/io/harness/cvng/analysis/services/impl/DeploymentTimeSeriesAnalysisServiceImpl.java#L467)):*
<details><summary><i>Previously, n00bitax (Ansuman Satapathy) wrote…</i></summary><blockquote>

too much code in this class, we should move all these static methods to a helper class.
</blockquote></details>

Will refactor
___
*[`srm-service/modules/cv-nextgen-service/contracts/commons/src/main/java/io/harness/cvng/cdng/beans/v2/ClusterAnalysisOverview.java` line 17 at r3](https://reviewable.io/reviews/harness/harness-core/42020#-NKMySEL77TuxfgEx_M8-r3-17:-NKRjQxO8hq4c9AMMEDB:bebrt1w) ([raw file](https://github.com/harness/harness-core/blob/4b5095f490cdefcc1340afbacacec95a76a09a87/srm-service/modules/cv-nextgen-service/contracts/commons/src/main/java/io/harness/cvng/cdng/beans/v2/ClusterAnalysisOverview.java#L17)):*
<details><summary><i>Previously, n00bitax (Ansuman Satapathy) wrote…</i></summary><blockquote>

my suggestion will be to suffix it with count. 
</blockquote></details>

Will do

**Review by n00bitax — APPROVED** [review]
> Reviewed 8 of 18 files at r1, 12 of 20 files at r2, 2 of 5 files at r3, 8 of 8 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42020)* status: 3. unresolved discussions.

**Review by n00bitax — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42020)* status: :shipit: complete! ✓

**Review by n00bitax — APPROVED** [review]

**Review by AbhijithHarness — APPROVED** [review]
> Reviewed 8 of 18 files at r1, 12 of 20 files at r2, 2 of 5 files at r3, 8 of 8 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42020)* status: :shipit: complete! ✓

**Review by n00bitax — APPROVED** [review]

**Review by n00bitax — APPROVED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42020)* status: :shipit: complete! ✓

**Review by AbhijithHarness — APPROVED** [review]
> Reviewed 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42020)* status: :shipit: complete! ✓

**Pull request merged**